ui/src/app/base/components/FabricSelect/FabricSelect.tsx

```diff
@@ -1,49 +1,51 @@
 import { useEffect } from "react";
 
-import { Spinner } from "@canonical/react-components";
+import { useFormikContext } from "formik";
 import { useDispatch, useSelector } from "react-redux";
 
 import DynamicSelect from "app/base/components/DynamicSelect";
 import type { Props as FormikFieldProps } from "app/base/components/FormikField/FormikField";
 import { actions as fabricActions } from "app/store/fabric";
 import fabricSelectors from "app/store/fabric/selectors";
-import type { Fabric } from "app/store/fabric/types";
 
 type Props = {
-  defaultOption?: { label: string; value: string } | null;
+  defaultOption?: { label: string; value: string; disabled?: boolean } | null;
 } & FormikFieldProps;
 
 export const FabricSelect = ({
-  defaultOption = { label: "Select fabric", value: "" },
+  defaultOption = { label: "Select fabric", value: "", disabled: true },
   name,
+  label = "Fabric",
+  disabled,
   ...props
 }: Props): JSX.Element => {
   const dispatch = useDispatch();
-  const fabrics: Fabric[] = useSelector(fabricSelectors.all);
+  const fabrics = useSelector(fabricSelectors.all);
   const fabricsLoaded = useSelector(fabricSelectors.loaded);
+  const { setFieldValue } = useFormikContext();
 
   useEffect(() => {
-    dispatch(fabricActions.fetch());
-  }, [dispatch]);
+    if (!fabricsLoaded) dispatch(fabricActions.fetch());
+  }, [dispatch, fabricsLoaded]);
 
-  if (!fabricsLoaded) {
-    return <Spinner />;
-  }
-
-  const fabricOptions = fabrics.map((fabric) => ({
-    label: fabric.name,
-    value: fabric.id.toString(),
-  }));
-
-  if (defaultOption) {
-    fabricOptions.unshift(defaultOption);
-  }
+  useEffect(() => {
+    if (fabricsLoaded) {
+      setFieldValue(name, fabrics[0]?.id?.toString());
+    }
+  }, [name, fabricsLoaded, fabrics, setFieldValue]);
```

Comment:
Seems like rendering a spinner until the fabrics has loaded was a workaround for the Formik Field not having a value if there are no options on initial render.

I'm setting a field value explicitly to the first item on load.

Reply:
This isn't quite accomplishing what you want it to. Any time the `fabrics` array changes the value is hard-set to the first in the array.

e.g open two tabs on the subnets page. On the first tab, open the Add VLAN form and select a fabric that isn't the first one and don't submit the form. On the second tab add a new fabric, and you should see the first tab's fabric selection is now the first one.

I think in general we shouldn't be using `setFieldValue` in a `useEffect` but rather as the result of some other explicit action, e.g. I change this select and then some other value changes too. Maybe this would work better if instead we just disable the field if there are no options?

Reply:
> Maybe this would work better if instead we just disable the field if there are no options?

That's what I intended - look below at `disabled={!fabricsLoaded || disabled}`